QuantLib/ql/termstructures/yield/fittedbonddiscountcurve.cpp

```diff
@@ -136,36 +144,49 @@ namespace QuantLib {
         Size n = curve_->bondHelpers_.size();
         costFunction_ = shared_ptr<FittingCost>(new FittingCost(this));
         costFunction_->firstCashFlow_.resize(n);
-        weights_ = Array(n);
-        Real squaredSum = 0.0;
-        for (Size i=0; i<curve_->bondHelpers_.size(); ++i) {
-            shared_ptr<Bond> bond = curve_->bondHelpers_[i]->bond();
-
-            Leg leg = bond->cashflows();
-            Real cleanPrice = curve_->bondHelpers_[i]->quote()->value();
-            
-            Date bondSettlement = bond->settlementDate();
-            Rate ytm = BondFunctions::yield(*bond, cleanPrice,
+        if(weights_.size() != n)
```

Comment:
This might probably be `if (weights_.empty())`. If the size is `!= n` but not null, we should probably raise an exception instead.